### PR TITLE
Log address and port, show exception trace from `uvicorn.run`

### LINKED ISSUE #562: [BUG] - conda-store-server silently exits when port is already in use
### Describe the bug

If you start conda-store-server when the configured port is already in use (e.g., because conda-store-server is already running), the process just silently exits. 

### Expected behavior

It should at least give an error message. 

Actually, it should probably do one of the following:

- Automatically find a free port.
- If it doesn't actually make sense for multiple conda-store servers to be running simultaneously, automatically check if an existing conda-store server is running somewhere and fail if it is. 

Even if the latter is the case, finding a free port could be useful in case some other application is using port 5000. 

### How to Reproduce the problem?

Run `conda-store-server` then run `conda-store-server` again in a different terminal. 

### Output

This is the only output that appears

```
$ conda-store-server
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
```

However, you can tell what the real problem is if you make the following patch

```diff
diff --git a/conda-store-server/conda_store_server/server/app.py b/conda-store-server/conda_store_server/server/app.py
index 741a31d..8f2dde7 100644
--- a/conda-store-server/conda_store_server/server/app.py
+++ b/conda-store-server/conda_store_server/server/app.py
@@ -363,6 +363,10 @@ class CondaStoreServer(Application):
                     else []
                 ),
             )
+        except:
+            import traceback
+            traceback.print_exc()
+            raise
         finally:
             if self.standalone:
                 process.join()
```

Then you get the error in the additional printed traceback:

```pytb
$ conda-store-server
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Traceback (most recent call last):
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 160, in startup
    server = await loop.create_server(
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 1493, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 48] error while attempting to bind on address ('0.0.0.0', 5000): address already in use

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/aaronmeurer/Documents/conda-store/conda-store-server/conda_store_server/server/app.py", line 352, in start
    uvicorn.run(
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/main.py", line 587, in run
    server.run()
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 628, in run_until_complete
    self.run_forever()
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 595, in run_forever
    self._run_once()
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 1881, in _run_once
    handle._run()
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 78, in serve
    await self.startup(sockets=sockets)
  File "/Users/aaronmeurer/anaconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 170, in startup
    sys.exit(1)
SystemExit: 1
```

### Versions and dependencies used.

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nkaretnikov**
Fixes #562.

## Description

This pull request:

- logs address and port on startup
- shows exception trace from `uvicorn.run`.

Example output when the port is already in use:

```
INFO  [app] Starting server on 0.0.0.0:8080
Traceback (most recent call last):
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 160, in startup
    server = await loop.create_server(
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 1493, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 98] error while attempting to bind on address ('0.0.0.0', 8080): address already in use

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/test/conda-store/conda-store-server/conda_store_server/server/app.py", line 371, in start
    uvicorn.run(
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/main.py", line 587, in run
    server.run()
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 628, in run_until_complete
    self.run_forever()
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 595, in run_forever
    self._run_once()
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/base_events.py", line 1881, in _run_once
    handle._run()
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 78, in serve
    await self.startup(sockets=sockets)
  File "/home/test/miniconda3/envs/conda-store-server-dev/lib/python3.10/site-packages/uvicorn/server.py", line 170, in startup
    sys.exit(1)
SystemExit: 1
```

## Pull request checklist

- [x] Did you test this change locally?
- [ ] Did you update the documentation (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

File: conda-store-server/conda_store_server/server/app.py [hunk 1]
```diff
@@ -362,6 +362,12 @@ def start(self):
             process.start()
 
         try:
+            # Note: the logger needs to be defined here for the output to show
+            # up, self.log doesn't work here either
+            logger = logging.getLogger("app")
+            logger.setLevel(self.log_level)
+            logger.info(f"Starting server on {self.address}:{self.port}")
```
Comment (nkaretnikov) [inline conda-store-server/conda_store_server/server/app.py:369]: We could move this to `initialize`, then we'll be able to use `self.log`. But it'll be far from where `uvicorn` is actually started.

Comment (nkaretnikov) [reply conda-store-server/conda_store_server/server/app.py:369]: This needs to be updated to use `basicConfig` because this uses the implicit handler.

**Comment by trallard** [comment]
ping @dcmcand for a review

**Review by dcmcand — APPROVED** [review]
I am a bit confused why self.log doesn't work in the start method while it does in the initialize method, but this seems to work and adds valuable info for people. :rocket:

**Pull request merged**